enzy_htp/_interface/pymol_interface.py

```diff
@@ -110,70 +227,266 @@ def config(self) -> PyMolConfig:
+        self.cmd.iterate('all', 'in_stru.append((name, elem, ID))', space=_eh_local)
+
+        if len(_eh_local['template']) != len(_eh_local['in_stru']):
+            _LOGGER.error(f"Atom number mismatch in template vs in_file structures. ({len(_eh_local['template'])} vs {len(_eh_local['in_stru'])}). Exiting...")
+            exit( 1 )
+        
+        for tidx, (tp1, tp2) in enumerate(zip(
+                sorted(_eh_local['template'], key=lambda tp: tp[-1]),
+                sorted( _eh_local['in_stru'], key=lambda tp: tp[-1])
+                 )):
+            if tp1[1] != tp2[1]:
+                _LOGGER.error(f"Atom type mismatch at atom number {tidx+1} template vs in_file structures. template: {tp1[1]}  in_stru: {tp2[1]}. Exiting...")
+                exit( 1 )
+
+        for (name, _, ID) in _eh_local['template']:
+            self.cmd.alter( f"ID {ID}", f'name="{name}"' )
+
+        self.cmd.save( in_file )
+        self.cmd.delete('all')
+
+        return in_file
+
+
+    def collect(self, molfile:str, variables:List[str], sele:str='all', state:int=-1, session:pymol2.PyMOL=None) -> pd.DataFrame:
+        """Function that serves as a wrapper to the pymol cmd API's iterate_state() method. Checks if the supplied file exists prior to collecting data.
+        Data is returned in a pandas DataFrame() where each row is an atom and each column is a variable.
+
+        Args:
+            molfile: Name of the structure to sample. 
+            variables: A list() of str()'s with variable names to pull from in the structure.
+            sele: The pymol selection string to use. Defaults to 'all'
+            state: The state to use. Defaults to -1 (current state).
+            session : A pymol2.PyMOL() session to use. By default uses an internal session instance that resets each function call.
+
+        Returns:
+            A pandas DataFrame() where each row is an atom and each column is a variable.
+
+        """
+        self.check_pymol2_installed()
+
+        if not session:
+            self.session_.cmd.delete('all')
+            return self.collect(
+                molfile=molfile,
+                variables=variables,
+                sele=sele,
+                state=state,
+                session=self.session_,
+            )
+
+
+        check_var_type(variables, list)
+        for vv in variables:
+            check_var_type( vv, str )
+
+        if molfile != 'memory':
+            fs.check_file_exists(molfile)    
+            session.cmd.delete('all')
+            session.cmd.load(molfile)
+
+        _eh_local: Dict[str, Any] = {'data':[]}
+
+        iter_stmt:str=f"data.append(({', '.join(variables)}))"
+        session.cmd.iterate_state(state,sele,iter_stmt,space=_eh_local)
+
+        result = pd.DataFrame(
+            data=_eh_local['data'],
+            columns=variables
+        )
+
+        return result
+
+    def execute(self, args:List[Tuple], session:pymol2.PyMOL=None) -> List[Any]:
```

Comment:
Could general_cmd() a better name. So that it is comparable with other functions in the interface that runs specific cmds.

Reply:
Sure, idrc.

Reply:
Changed in a commit.